### PR TITLE
Save& File - Unit testing & completeFiling Functionality

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#1896

*Description of changes:*

This code is an update to the frontend regarding some new information about the endpoints.
Essentially the need for a `draft` parameter is no longer required, also the complete filing will be a 2 step request to ensure there is a `filingId` even if the user goes through the flow in one shot without saving. 

* Implemented Unit testing
* updated competeFilings functionality

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/common/Actions.vue [hunk 1]
```diff
@@ -122,7 +122,7 @@ export default class Actions extends Mixins(FilingTemplateMixin, LegalApiMixin)
   private async onClickSaveResume (): Promise<void> {
     this.setIsSavingResuming(true)
     const filing = await this.buildFiling()
-    await this.saveFiling(filing, true)
+    await this.saveFiling(filing)
```
Comment (severinbeauvais) [inline src/components/common/Actions.vue]: I think you should wrap this statement in a try-catch block, and unwrap the methods in legal-api-mixin. Then here you can display some error dialog. (Error dialogs are future so you could just alert() here for now.)
Basically, I think Actions.vue should display the error, not the mixin.

File: src/components/common/Actions.vue [hunk 2]
```diff
@@ -122,7 +122,7 @@ export default class Actions extends Mixins(FilingTemplateMixin, LegalApiMixin)
...
     this.setIsSavingResuming(false)
     window.location.assign(this.authUrl)
```
Comment (severinbeauvais) [inline src/components/common/Actions.vue]: Don't redirect if the save failed :)

File: src/components/common/Actions.vue [hunk 3]
```diff
@@ -134,7 +134,7 @@ export default class Actions extends Mixins(FilingTemplateMixin, LegalApiMixin)
   private async onClickFilePay (): Promise<void> {
     this.setIsFilingPaying(true)
     const filing = await this.buildFiling()
-    await this.saveFiling(filing, false)
+    await this.completeFiling(filing)
```
Comment (severinbeauvais) [inline src/components/common/Actions.vue]: If we do error handling in this component, do we need to splits up completeFiling into 2 separate calls -- the first to save and the second to file?

File: src/components/common/Actions.vue (hunk 1 above)
Comment (cameron-freshworks) [reply src/components/common/Actions.vue]: Makes sense, much easier to handle in the component itself than the mixin, will do!

File: src/components/common/Actions.vue (hunk 2 above)
Comment (cameron-freshworks) [reply src/components/common/Actions.vue]: kk, will do.

File: src/components/common/Actions.vue (hunk 3 above)
Comment (cameron-freshworks) [reply src/components/common/Actions.vue]: I'll look into it!

File: src/components/common/Actions.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/components/common/Actions.vue]: Assumption is that we'd `return` inside the catch block.

**Pull request merged**